popde/adaptive_kde.py

```diff
@@ -117,6 +117,100 @@ def set_adaptive_parameter(self, new_alpha, new_global_bw):
         self.set_alpha(new_alpha)
 
 
+class KDEOptimization(AdaptiveBwKDE):
+    """
+    Optimize bandwidth and alpha by grid search using
+    cross validation with a log likelihood figure of merit 
+    """
+    def __init__(self, data, bandwidth_options, alpha_options, weights=None, input_transf=None, stdize=False, rescale=None, backend='KDEpy', bandwidth=1.0, alpha=0.0, dim_names=None, do_fit=True, n_splits=2):
+        self.alpha_options = alpha_options
+        self.bandwidth_options = bandwidth_options
+        self.n_splits = n_splits
+        super().__init__(data, weights, input_transf, stdize,
+                         rescale, backend, bandwidth, alpha, dim_names, do_fit)
+
+    def loo_cv_score(self, bandwidth_val, alpha_val):
+        from sklearn.model_selection import LeaveOneOut
+        loo = LeaveOneOut() #sklearn way
+        fom = 0.0
+        for train_index, test_index in loo.split(self.data):
+            train_data, test_data = self.data[train_index], self.data[test_index]
+            #for weights we need corressponding weights as well
+            if self.weights is not None:
+                local_train_weights = weights[train_index]
+            local_weights = None # FIX ME
+            awkde = AdaptiveBwKDE(train_data, local_weights, bandwidth=bandwidth_val,alpha=alpha_val)
```

Comment:
I am not seeing what the line with local_train_weights is for (also in the kfold case). If the code does nothing at the moment then it is best not to write code at all but instead leave comments discussing possible improvements

Reply:
preparing kde with weights  need weight at the training points.  maybe important or disregard for optimized parameter, in progress

Reply:
but are you saying that you are going to include something to do this in this PR, or not? 
I am saying we should not have 'in progress' code that does not do anything in the repo.
so can we please remove the lines about `local_train_weights` and make notes for any relevant development in another issue or another PR?